agent/provider/src/tasks/task_manager.rs

```diff
@@ -165,13 +170,23 @@ impl TaskManager {
             }
         });
 
-        // Schedule agreement termination when there is no activity created within 90s.
-        let s90 = Duration::seconds(90).to_std()?;
-        ctx.run_later(s90.clone(), move |myself, ctx| {
+        self.schedule_idle_expiration(ScheduleIdleExpiration(msg.0), ctx)
+    }
+
+    fn schedule_idle_expiration(
+        &mut self,
+        msg: ScheduleIdleExpiration,
+        ctx: &mut Context<Self>,
+    ) -> Result<()> {
+        let idle_timeout = msg.0.idle_agreement_timeout;
+        let agreement_id = msg.0.agreement_id;
+
+        // Schedule agreement termination when there is no activity created within timeout.
+        ctx.run_later(idle_timeout.clone(), move |myself, ctx| {
```

Comment:
I'm not sure it is enough. What about the case when activity is started and finished between now and now + idle_timeout.

Reply:
Thanks for good catch